source/splitWith.js

```diff
@@ -0,0 +1,38 @@
+import _curry2 from './internal/_curry2';
+import _reduce from './internal/_reduce';
+
+
+/**
+ * Takes a list and a predicate and returns a list of lists split by matches
+ *
+ * @func
+ * @memberOf R
+ * @since v0.25.0
+ * @category List
+ * @sig (a -> Boolean) -> [a] -> [[a]]
+ * @param {Function} pred The predicate that determines where the array is split.
+ * @param {Array} list The array to be split.
+ * @return {Array}
+ * @example
+ *
+ *      R.splitWith(R.equals('a'), [1, 2, 3, 1, 2, 3]); //=> [[1], [3, 1], [3]]
+ */
+var splitWith = _curry2(function splitWith(pred, list) {
+  let currIndex = 0;
+  return _reduce(function(acc, curr) {
+    if (pred(curr)) {
+      if (currIndex !== list.length - 1) {
+        acc.push([]);
+      }
```

Comment:
This only works if there is a single matching value.  With this implementation,

```js
splitWith(equals(2), [2, 2, 1, 2, 3, 4, 1, 2, 2]) //=> [[], [1], [3, 4, 1], []]
```

So something needs to change.